platform/ui/src/components/MeasurementTable/MeasurementItem.tsx

```diff
@@ -24,6 +25,16 @@ const MeasurementItem = ({
   const onMouseEnter = () => setIsHovering(true);
   const onMouseLeave = () => setIsHovering(false);
 
+  const { site, finding } = item;
+  const useLabel = label || site?.text || finding?.text || '(empty)';
+  let useDisplayText = displayText;
+  if (finding && finding?.text !== useLabel) {
+    useDisplayText = [finding.text, ...useDisplayText];
+  }
+  if (site && site?.text !== useLabel) {
+    useDisplayText = [site.text, ...useDisplayText];
+  }
```

Comment:
I don't think these belong here, these logics should be a level up, and providing displayText and label as before, so that MeasurementItem becomes untouched

Reply:
Fixed